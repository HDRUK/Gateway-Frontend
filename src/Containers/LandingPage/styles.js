import styled from "styled-components";

export const LandingPageWrapper = styled.div`
    padding: 0 3rem 9rem 3rem;
`;
