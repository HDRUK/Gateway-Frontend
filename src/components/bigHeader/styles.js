import styled from "styled-components";
import { HeaderWrapper } from "../../styles/carbonComponents.js";

export const BigHeaderWrapper = styled(HeaderWrapper)`
    display: block;
    height: 33.5rem;
    padding-top: 2rem;
`;

export const BigHeaderImage = styled.img`
    height: 6.4375rem;
    display: block;
`;
