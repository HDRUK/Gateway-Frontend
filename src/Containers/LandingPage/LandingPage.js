import React from "react";
import NewsTile from "../../components/newsTile/newsTile.js";
import NewsTileGroup from "../../components/newsTileGroup/newsTileGroup.js";
import ImageBlock from "../../components/imageBlock/imageBlock.js";
import Image from "../../components/image/image.js";
import ParagraphBanner from "../../components/paragraphBanner/paragraphBanner.js";
import mainImage from "../../assets/landing_page_image_1.png";

import { LandingPageWrapper, ParagraphIcon, PartnerImage } from "./styles.js";
import {
    Heading2,
    ParagraphHeading,
    ParagraphHeading3,
    DarkText,
    SmallInlineSpace,
    SmallSpace,
    MediumSpace,
    ParagraphText,
    ParagraphTextHero,
    ParagraphAndHeaderBox,
    ParagraphBox,
    MainImage,
    CenterBlock
} from "../../styles/styles.js";

import { Search32, Send32, Locked32 } from "@carbon/icons-react";

const textItems = {
    landingFirstParaHeading: "What is the Innovation Gateway?",
    landingFirstPara:
        "Welcome to the Health Data Research Innovation Gateway. This is a portal to find and enquire about access to UK health datasets for research. The Innovation Gateway provides detailed descriptions of these datasets, which you can search, browse and enquire about access. It does not hold or store any patient or health data.",
    landingSecondPara: "The Innovation Gateway does not hold or store any patient or health data.",
    bannerFirstTitle: "Discover",
    bannerFirstPara:
        "The Innovation Gateway allows you to quickly discover and identify health related datasets. Use a keyword search to get detailed descriptions of the datasets that interest you.  You can sort, filter and search results to find what you need.",
    bannerSecondTitle: "Enquire",
    bannerSecondPara:
        "You need to be a researcher or innovator affiliated to a known organisation to enquire about access to the datasets. Sign in via OpenAthens to enquire about access to the datasets that interest you.",
    bannerThirdTitle: "Security",
    bannerThirdPara:
        "All data access enquires are sent to the relevant data custodian(s) for consideration.  The Innovation Gateway supports members of the UK Health Data Research Alliance, who have subscribed to the secure safeguarding of personal health data and responsible accessto this data for research and innovation.",
    partnersHeading: "Alliance Members include"
};

const LandingPage = () => {
    return (
        <LandingPageWrapper>
            <MediumSpace />
            <DarkText>
                <ParagraphAndHeaderBox>
                    <Heading2>{textItems.landingFirstParaHeading}</Heading2>
                    <SmallSpace />
                    <ParagraphBox>
                        <ParagraphTextHero>{textItems.landingFirstPara}</ParagraphTextHero>
                        <SmallSpace />
                        <ParagraphText>{textItems.landingSecondPara}</ParagraphText>
                    </ParagraphBox>
                    <SmallSpace />
                </ParagraphAndHeaderBox>
                <SmallInlineSpace />
                <MainImage src={mainImage} />
            </DarkText>
            <MediumSpace />
            <ParagraphBanner>
                <>
                    <ParagraphIcon>
                        <Search32 />
                    </ParagraphIcon>
                    <ParagraphAndHeaderBox>
                        <ParagraphHeading3>{textItems.bannerFirstTitle}</ParagraphHeading3>
                        <SmallSpace />
                        <ParagraphBox>
                            <ParagraphText>{textItems.bannerFirstPara}</ParagraphText>
                        </ParagraphBox>
                    </ParagraphAndHeaderBox>
                </>

                <>
                    <ParagraphIcon>
                        <Send32 />
                    </ParagraphIcon>
                    <ParagraphAndHeaderBox>
                        <ParagraphHeading3>{textItems.bannerSecondTitle}</ParagraphHeading3>
                        <SmallSpace />
                        <ParagraphBox>
                            <ParagraphText>{textItems.bannerSecondPara}</ParagraphText>
                        </ParagraphBox>
                    </ParagraphAndHeaderBox>
                </>

                <>
                    <ParagraphIcon>
                        <Locked32 />
                    </ParagraphIcon>
                    <ParagraphAndHeaderBox>
                        <ParagraphHeading3>{textItems.bannerThirdTitle}</ParagraphHeading3>
                        <SmallSpace />
                        <ParagraphBox>
                            <ParagraphText>{textItems.bannerThirdPara}</ParagraphText>
                        </ParagraphBox>
                    </ParagraphAndHeaderBox>
                </>
            </ParagraphBanner>
            <MediumSpace />
            <NewsTileGroup>
                <NewsTile identifier="newsItemOne" />
                <NewsTile identifier="newsItemTwo" />
                <NewsTile identifier="newsItemThree" />
            </NewsTileGroup>
            <MediumSpace />
            <CenterBlock>
                <ParagraphHeading>{textItems.partnersHeading}</ParagraphHeading>
            </CenterBlock>
            <SmallSpace />
            <ImageBlock>
                <PartnerImage>
                    <Image identifier="partnerLogo1" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo2" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo3" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo4" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo5" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo6" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo7" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo8" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo9" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo10" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo11" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo12" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo13" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo14" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo15" />
                </PartnerImage>
                <PartnerImage>
                    <Image identifier="partnerLogo16" />
                </PartnerImage>
            </ImageBlock>
        </LandingPageWrapper>
    );
};

export default LandingPage;
