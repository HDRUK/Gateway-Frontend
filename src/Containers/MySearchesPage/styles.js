import styled from "styled-components";
import { ResultsWrapper } from "../../styles/styles.js";

export const SavedSearchesWrapper = styled(ResultsWrapper)`
    height: ${p => (p.visible ? "calc(100vh - 8rem)" : "calc(100vh - 21rem)")};
    ${p => p.visible && "max-height: calc(100vh - 8rem);"};
`;
