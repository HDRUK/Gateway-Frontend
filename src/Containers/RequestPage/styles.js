import styled from "styled-components";
import { NewStyledButton, RightSmallInlineLoading } from "../../styles/carbonComponents";

export const LoadingButton = styled(NewStyledButton)`
    width: 11rem;
    display: flex;
    justify-content: center;
`;

export const LoadingWheel = styled(RightSmallInlineLoading)`
    margin-top: -0.4rem;
`;
