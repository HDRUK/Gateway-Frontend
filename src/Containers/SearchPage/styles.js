import styled from "styled-components";
import { ParagraphHeading, ResultsWrapper } from "../../styles/styles.js";

export const SearchHeading = styled(ParagraphHeading)`
    height: 0;
    text-align: center;
    transform: translateY(12rem);
    opacity: ${p => (p.invisible ? 0 : 1)};
    transition: 0.25s;
`;

export const SearchBarWrapper = styled.div`
    max-width: 32rem;
    padding: 0 1rem;
    margin: 0 auto 0;
    text-align: center;
    transform: translateY(${p => (p.main ? "16rem" : "0")});
    transition: 0.7s;
`;

export const Results = styled.div`
    opacity: ${p => (p.invisible ? 0 : 1)};
    transition: 0.8s;
    transition-delay: 0.5s;
    pointer-events: ${p => (p.invisible ? "none" : "unset")};
`;

export const SearchResultsWrapper = styled(ResultsWrapper)``;
