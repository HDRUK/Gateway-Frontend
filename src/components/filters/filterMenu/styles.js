import styled from "styled-components";
import { AccordionBlock } from "../../../styles/carbonComponents";

export const ExpandingAccordionBlock = styled(AccordionBlock).attrs(p => ({
    style: {
        paddingBottom: p.expandheight ? `${p.expandheight}px` : "1rem"
    }
}))`
    display: inline-block;
    padding-bottom: "1rem";
`;
