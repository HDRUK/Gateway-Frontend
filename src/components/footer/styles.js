import styled from "styled-components";
import { WidthWrapper, Logo } from "../../styles/styles.js";

export const FooterWrapper = styled.div`
    height: ${p => p.theme.sizes.footer}rem;
    background: linear-gradient(135deg, ${p => p.theme.colors.primary}, ${p => p.theme.colors.secondary});
    color: white;
    padding-top: 4rem;
    padding-bottom: 1rem;
    font-size: 1.1rem;
    position: relative;
`;

export const FooterContent = styled(WidthWrapper)`
    display: block;
    margin: 0 auto 0;
    max-width: 62rem;
    padding: 0 2rem;
`;

export const FooterBlock = styled.div`
    display: inline-block;
    box-sizing: border-box;
    width: 33.33%;
    height: 100%;
    vertical-align: top;
    padding: 0 4rem 0 0;
    line-height: 1.1875rem;

    :last-of-type {
        padding-right: 0;
        padding-left: 4rem;
    }
`;

export const FooterText = styled.div`
    width: 100%;
    margin: 0 0 0.4rem 0;
`;

export const FooterImage = styled(Logo)`
    max-width: 8rem;
    max-height: 5rem;
`;
