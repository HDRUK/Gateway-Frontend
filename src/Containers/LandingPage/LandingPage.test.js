import React from "react";
import { create } from "react-test-renderer";
import LandingPage from "./LandingPage.js";
import { BrowserRouter } from "react-router-dom";

import { AppContext } from "../../HOC/AppContext/AppContext.js";
import context from "../../__mocks__/AppContextMock.js";

import Paragraph from "../../components/paragraph/paragraph.js";
import Login from "../../components/login/login.js";
import NewsTile from "../../components/newsTile/newsTile.js";
import NewsTileGroup from "../../components/newsTileGroup/newsTileGroup.js";
import ImageBlock from "../../components/imageBlock/imageBlock.js";

import { ParagraphHeading, DarkText, NewListItem, LinkText, MediumSpace, SmallSpace } from "../../styles/styles.js";
import { ParagraphBullets } from "../../styles/carbonComponents.js";

describe("<LandingPage> ", () => {
    let renderedComponent;

    beforeEach(() => {
        renderedComponent = create(
            <AppContext.Provider value={context}>
                <BrowserRouter>
                    <LandingPage />
                </BrowserRouter>
            </AppContext.Provider>
        );
    });

    it("should test the LandingPage Snapshot", () => {
        expect(renderedComponent).toMatchSnapshot();
    });

    it("should check the crrect components are displayed", () => {
        const paragraphs = renderedComponent.root.findAllByType(Paragraph);
        const login = renderedComponent.root.findAllByType(Login);
        const newsTiles = renderedComponent.root.findAllByType(NewsTile);
        const newsTileGroup = renderedComponent.root.findAllByType(NewsTileGroup);
        const imageBlock = renderedComponent.root.findAllByType(ImageBlock);
        const paragraphHeadings = renderedComponent.root.findAllByType(ParagraphHeading);
        const paragraphBullets = renderedComponent.root.findAllByType(ParagraphBullets);
        const darkTexts = renderedComponent.root.findAllByType(DarkText);
        const smallSpaces = renderedComponent.root.findAllByType(SmallSpace);
        const mediumSpaces = renderedComponent.root.findAllByType(MediumSpace);
        const newListsItems = renderedComponent.root.findAllByType(NewListItem);
        const linkTexts = renderedComponent.root.findAllByType(LinkText);

        expect(paragraphs).toHaveLength(3);
        expect(login).toHaveLength(1);
        expect(newsTiles).toHaveLength(3);
        expect(newsTileGroup).toHaveLength(1);
        expect(imageBlock).toHaveLength(1);
        expect(paragraphHeadings).toHaveLength(4);
        expect(paragraphBullets).toHaveLength(2);
        expect(darkTexts).toHaveLength(6);
        expect(smallSpaces).toHaveLength(8);
        expect(mediumSpaces).toHaveLength(3);
        expect(newListsItems).toHaveLength(6);
        expect(linkTexts).toHaveLength(3);
    });
});
