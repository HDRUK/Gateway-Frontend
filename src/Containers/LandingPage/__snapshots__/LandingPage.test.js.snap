// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage>  should test the LandingPage Snapshot 1`] = `
Array [
  <div
    className="sc-iwsKbI kOWQtT"
  />,
  <div
    className="sc-htoDjs dtAbCu"
  >
    <text
      className="sc-EHOje cUVptH"
    >
      <h1
        className="sc-bdVaJa hSReXH"
      >
        Log in to access our datasets
      </h1>
      <div
        className="sc-iwsKbI kOWQtT"
      />
      <div
        className="sc-htoDjs dtAbCu"
      >
        <button
          className="sc-hSdWYo kgbWYs bx--btn bx--btn--primary"
          disabled={false}
          tabIndex={0}
          type="button"
        >
          Log in
        </button>
        <div
          className="sc-dnqmqq dlGTFF"
        />
        <a
          href="/search"
          onClick={[Function]}
        >
          <button
            className="sc-hSdWYo kgbWYs bx--btn bx--btn--secondary"
            disabled={false}
            tabIndex={0}
            type="button"
          >
            Continue without logging in
          </button>
        </a>
      </div>
      <div
        className="sc-gZMcBi jYAjms"
      />
    </text>
  </div>,
  <text
    className="sc-EHOje cUVptH"
  >
    <h2
      className="sc-bwzfXH gshKuR"
    >
      What is the Innovation Gateway
    </h2>
    <div
      className="sc-dnqmqq dlGTFF"
    />
    <text
      className="sc-EHOje cUVptH"
    >
      <p
        className="sc-ifAKCX hSPtPb"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </text>
  </text>,
  <div
    className="sc-dnqmqq dlGTFF"
  />,
  <div
    className="sc-kEYyzF dPKAJW"
  >
    <img
      className="sc-kkGfuU dsSJCT"
      src="hdruk_black.png"
    />
    <img
      className="sc-kkGfuU dsSJCT"
      src="hdruk_black.png"
    />
    <img
      className="sc-kkGfuU dsSJCT"
      src="hdruk_black.png"
    />
  </div>,
  <div
    className="sc-dnqmqq dlGTFF"
  />,
  <h2
    className="sc-bwzfXH gshKuR"
  >
    HDR News
  </h2>,
  <div
    className="sc-dnqmqq dlGTFF"
  />,
  <div>
    <div
      className="sc-iAyFgw dMPTmX"
    >
      <div
        className="sc-dxgOiQ eJgKXn"
      >
        <div
          className="sc-ckVGcZ ObcPe"
          src=""
        />
        <div
          className="sc-jKJlTe kCudHY"
        >
          <div
            className="sc-eNQAEJ cuBSny"
          >
            HDR release news of new partners in exciting new digital project
          </div>
          <a
            className="sc-hMqMXs jGyQhK"
            href="http://localhost:3000"
          >
            Read more
          </a>
        </div>
      </div>
    </div>
    <div
      className="sc-iAyFgw dMPTmX"
    >
      <div
        className="sc-dxgOiQ eJgKXn"
      >
        <div
          className="sc-ckVGcZ ObcPe"
          src=""
        />
        <div
          className="sc-jKJlTe kCudHY"
        >
          <div
            className="sc-eNQAEJ cuBSny"
          >
            HDR release news of new partners in exciting new digital project
          </div>
          <a
            className="sc-hMqMXs jGyQhK"
            href="http://localhost:3000"
          >
            Read more
          </a>
        </div>
      </div>
    </div>
    <div
      className="sc-iAyFgw dMPTmX"
    >
      <div
        className="sc-dxgOiQ eJgKXn"
      >
        <div
          className="sc-ckVGcZ ObcPe"
          src=""
        />
        <div
          className="sc-jKJlTe kCudHY"
        >
          <div
            className="sc-eNQAEJ cuBSny"
          >
            HDR release news of new partners in exciting new digital project
          </div>
          <a
            className="sc-hMqMXs jGyQhK"
            href="http://localhost:3000"
          >
            Read more
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-dnqmqq dlGTFF"
  />,
  <text
    className="sc-EHOje cUVptH"
  >
    <h2
      className="sc-bwzfXH gshKuR"
    >
      Guidelines for using the Innovation Gateway
    </h2>
    <div
      className="sc-dnqmqq dlGTFF"
    />
    <text
      className="sc-EHOje cUVptH"
    >
      <p
        className="sc-ifAKCX hSPtPb"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </p>
    </text>
    <ul
      className="bx--list--unordered sc-eHgmQL bPaHwg"
    >
      <li
        className="sc-bZQynM hRgGJr"
      >
        Lorem ipsum
      </li>
      <li
        className="sc-bZQynM hRgGJr"
      >
        Lorem ipsum
      </li>
      <li
        className="sc-bZQynM hRgGJr"
      >
        Lorem ipsum
      </li>
    </ul>
    <div
      className="sc-iwsKbI kOWQtT"
    />
    <h2
      className="sc-bwzfXH gshKuR"
    >
      Help using the Innovation Gateway
    </h2>
    <div
      className="sc-dnqmqq dlGTFF"
    />
    <text
      className="sc-EHOje cUVptH"
    >
      <p
        className="sc-ifAKCX hSPtPb"
      >
        There are several ways to get help using the Innovation Gateway
      </p>
    </text>
  </text>,
  <ul
    className="bx--list--unordered sc-eHgmQL bPaHwg"
  >
    <li
      className="sc-bZQynM hRgGJr"
    >
      <a
        className="sc-gzVnrw hGlsen"
      >
        Frequently asked questions
      </a>
    </li>
    <li
      className="sc-bZQynM hRgGJr"
    >
      <a
        className="sc-gzVnrw hGlsen"
      >
        Forum
      </a>
    </li>
    <li
      className="sc-bZQynM hRgGJr"
    >
      <a
        className="sc-gzVnrw hGlsen"
      >
        Get in touch with HDR UK
      </a>
    </li>
  </ul>,
]
`;
